tests/ocf-url_parse_leaking_relative/EPUB/content_001.xhtml

```diff
@@ -0,0 +1,60 @@
+<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
+    <head>
+        <meta charset="utf-8"/>
+        <title>Parse a potentially leaking URL</title>
+        <script type="text/javascript">
+            const expected_path   = 'media/imgs/monastery.jpg';
+            const parent_relative = `../../../../../${expected_path}`;
+            const current_path    = 'EPUB/content_001.xhtml'
+            function test_url() {
+                try {
+                    document.getElementById('no_scripting').style.display = 'none';
+                    document.getElementById('scripting').style.display = 'block';
+
+                    // in some cases this URL contains search or fragment, this should be removed to get to the
+                    // 'pure' URL. This is done by cloning the document's URI and removing  
+                    // relevant portions of the full URL
+                    // (There are RS-s that add a long query string to the URL, hence the necessity of this step)
+                    const new_URL = new URL(document.documentURI);
+                    new_URL.search = '';  new_URL.hash = '';
+                    const documentURI = new_URL.toString();
+
+                    // This is, hopefully, a good representation of the container root URL: remove the
+                    // relative path of the current document
+                    const container_root_URL = documentURI.slice(0,documentURI.length - current_path.length);
+                    document.getElementById('container_root_url').textContent = container_root_URL;
+
+                    document.getElementById('test_url').textContent = parent_relative;
+
+                    const image_src = new URL(parent_relative, container_root_URL);
+                    document.getElementById('image_url_parent_paths').textContent = image_src;
+
+                    document.getElementById('expected').textContent = `{Container Root URL}${expected_path}`;
+                } catch(e) {
+                    alert(e)
+                }
+            }
+            window.addEventListener('load', test_url);
+        </script>
+    </head>
+    <body>
+        <p id="no_scripting" style="display: block">This reading system does not support scripting, which is necessary for this test to run.</p>
```

Comment:
it's a detail and not specific to the test itself: but this could probably rely on HTML's `noscript` element?

Reply:
When I first read the comment, my reaction was "Oops...". But then I got suspicious and found this:

> The [noscript](https://html.spec.whatwg.org/multipage/scripting.html#the-noscript-element) element is only effective in [the HTML syntax](https://html.spec.whatwg.org/multipage/syntax.html#syntax), it has no effect in [the XML syntax](https://html.spec.whatwg.org/multipage/xhtml.html#the-xhtml-syntax). This is because the way it works is by essentially "turning off" the parser when scripts are enabled, so that the contents of the element are treated as pure text and not as real elements. XML does not define a mechanism by which to do this.

See [HTML spec](https://html.spec.whatwg.org/multipage/scripting.html#the-noscript-element). I guess that means we should not use it in EPUB...

Reply:
Oh, TIL. Good catch!